### ISSUE TITLE
lychee ignores http accepted status codes when using cache

### ISSUE THREAD

**Issue by ruzickap** [lycheeverse/lychee#840]
It looks like lychee doesn't work with cache and accepted status codes properly.

See the example:

Prepare the data + directory:

```bash
mkdir /tmp/test
cd /tmp/test

cat > lychee.toml << EOF
accept = [ 999 ]
EOF

cat > README.md << EOF
[https://www.linkedin.com/in/petrruzicka/](https://www.linkedin.com/in/petrruzicka/)
EOF
```

Run lychee (first run is working fine):

```bash
$ lychee --cache .
⠉ 1/2 [00:00:00] ████████████░░░░░░░░░░░░░ ✔ [999] https://www.linkedin.com/in/petrruzicka/ | OK (999 <unknown status co
⠉ 2/2 [00:00:00] █████████████████████████ ✔ [999] https://www.linkedin.com/in/petrruzicka/ | OK (999 <unknown status co
🔍 2 Total ✅ 2 OK 🚫 0 Errors
```

Second run is failing - I still have `accept = [ 999 ]` in `lychee.toml` :-(

```bash
$ lychee --cache .
Issues found in 1 input. Find details below.

[./README.md]:
↻ [999] https://www.linkedin.com/in/petrruzicka/ | Cached: Error (cached)

🔍 2 Total ✅ 0 OK 🚫 2 Errors (HTTP:2)
```

Cache details:

```bash
❯ cat .lycheecache
https://www.linkedin.com/in/petrruzicka/,999,1669292835
```

Seems to be a bug...

Thank you for looking at it.

**Comment by mre** [comment]
Yup, that's a bug. Thanks for reporting.

**Comment by mre** [comment]
Fixed in `master`. Will be released with next version. Thanks for the report @ruzickap.

**Comment by ruzickap** [comment]
Thanks a lot :-)

**Comment by t1m0thyj** [comment]
@mre Thanks for fixing this, will a new version of Lychee be released soon? 🙏 This change would be very useful for my team, since the caching helps us to avoid rate limiting, but we also need to ignore some status codes.

**Comment by mre** [comment]
Hey @t1m0thyj, sure. Just released lychee 0.11.0 and lychee-action 1.6.0, which include the change.

**Comment by mre** [comment]
FYI: I had to do another hotfix release 0.11.1 to fix a regression with handling cached 2xx status codes.
See #958.
